### PR TITLE
feat(elements): Initial virtual router

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

Fixes SDK-1789

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by tmilewski** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @tmilewski - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.17-snapshot.v93b4914 |
| @clerk/clerk-js | 5.5.4-snapshot.v93b4914 |
| @clerk/clerk-expo | 1.1.9-snapshot.v93b4914 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.4-snapshot.v93b4914 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.17-snapshot.v93b4914 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.5.4-snapshot.v93b4914 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.9-snapshot.v93b4914 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.4-snapshot.v93b4914 --save-exact
```

**Comment by clerk-cookie** [comment]
Hey @tmilewski - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.17-snapshot.vec69ef6 |
| @clerk/clerk-js | 5.5.4-snapshot.vec69ef6 |
| @clerk/elements | 0.5.3-snapshot.vec69ef6 |
| @clerk/clerk-expo | 1.1.9-snapshot.vec69ef6 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.4-snapshot.vec69ef6 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.17-snapshot.vec69ef6 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.5.4-snapshot.vec69ef6 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.5.3-snapshot.vec69ef6 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.9-snapshot.vec69ef6 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.4-snapshot.vec69ef6 --save-exact
```

**Comment by tmilewski** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @tmilewski - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.17-snapshot.v3c4cd45 |
| @clerk/clerk-js | 5.5.4-snapshot.v3c4cd45 |
| @clerk/elements | 0.6.0-snapshot.v3c4cd45 |
| @clerk/clerk-expo | 1.1.9-snapshot.v3c4cd45 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.4-snapshot.v3c4cd45 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.17-snapshot.v3c4cd45 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.5.4-snapshot.v3c4cd45 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.6.0-snapshot.v3c4cd45 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.9-snapshot.v3c4cd45 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.4-snapshot.v3c4cd45 --save-exact
```

**Review by BRKalow — APPROVED** [review]
👍

File: packages/elements/src/react/sign-in/root.tsx [hunk 1]
```diff
@@ -80,22 +79,23 @@ export type SignInRootProps = {
...
-      fallback: Boolean(fallback),
       exampleMode: Boolean(exampleMode),
+      fallback: Boolean(fallback),
+      path,
     }),
   );
 
   // TODO: eventually we'll rely on the framework SDK to specify its host router, but for now we'll default to Next.js
-  const router = useNextRouter();
+  const router = (routing === ROUTING.virtual ? useVirtualRouter : useNextRouter)();
```
[the remark this answers is not in the document]
Comment (tmilewski) [reply packages/elements/src/react/sign-in/root.tsx:98]: Hahaha

image

**Pull request merged**